### DISCUSSION TITLE
[Question][Hw3] How is grading done / More test case

### DISCUSSION THREAD

**Discussion by Standchen** [prosyslab-classroom/cs348-information-security#511] [Q&A]
Hello.
In assignment3, I'd like to ask how is grading done after we submit our work.
It is more likely to believe that the grading will be done with more test case, not just only with the single test case we provided in the repo.

If the grading will be done with additional test case, can we get supplemented more test case for the improvement to our work?

**Comment by KihongHeo** [comment]
Yes. We will use a test-case-based grading.
Once all the grading is done, TAs will push the used test cases to each of your repository so that you can verify the score by yourself.

**Comment by Standchen** [comment]
> 
> 
> Yes. We will use a test-case-based grading.
> Once all the grading is done, TAs will push the used test cases to each of your repository so that you can verify the score by yourself.

@KihongHeo Thank you for replying my question. If the grading is test-case-based, then can we get a few more test cases, to check our progress in doing assignment? Currently, provided test case is only single, very rudimentary one, which doesn't clarify the requirements.
I think we need to handle the edge cases for coverage(function call, binary op, unary op, phi, etc.), but the document doesn't specify such criteria, and the number of test case is insufficient to verify whether our implementation met such conditions.

In short: Can you provide us more test cases, to reduce ambiguity and provide guidelines for implementation?

**Comment by KihongHeo** [comment]
Originally, I intentionally put one test case because of the following reasons:
- I want to encourage students to make their own test cases
- I want students to think about more fundamental, general principles before starting implementation
- I think this homework does not have much ambiguity and many edge cases

I have asked TAs to provide two more test cases. But please notice my points above.
